### ISSUE TITLE
numpy.roots() does not return roots if they are all zero. (migrated from Trac #401)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#1008]
Original ticket http://projects.scipy.org/numpy/ticket/401
Reported 2006-12-12 by trac user DaleD, assigned to unknown.

Python 2.4.3, Enthought Edition 1.0.0, on Windows, numpy.version.version=
'0.9.9.2706'

If all of the roots of a polynomial are zero, then an empty array is returned from roots().  Example:

> > > from numpy import *
> > > roots(array([1.0,0,1.0]))
> > > array([ 0.+1.j,  0.-1.j])      # correct
> > > roots(array([1.0,0,0]))
> > > array([], dtype=int32)         # either incorrect or inconvenient
> > > roots(array([1.0,1.0,0]))
> > > array([-1.,  0.])              # returns the zero root okay here

Should either document this behavior(yuck) or return all the zero roots(cool).

**Comment by thouis** [comment]
Comment in Trac by atmention:nilswagner01, 2006-12-12

This is fixed in svn.

```
>>> from numpy import *
>>> roots(array([1.0,0,0]))
array([ 0.,  0.])
>>> import numpy
>>> numpy.__version__
'1.0.2.dev3483'
```

**Comment by thouis** [comment]
Comment in Trac by atmention:timleslie, 2007-01-08

This was fixed in [3265](see #323).
